### PR TITLE
createClass + PropTypes + checkPropTypes warnings

### PR DESCRIPTION
**Pull request by bvaughn**
This PR is prep work for a new 16 alpha release that (temporarily) re-adds getters with deprecation warnings for `React.PropTypes`, `React.checkPropTypes`, and `React.createClass`. I'm hoping to get this new alpha cherry picked into RN 44.

I've broken this PR into multiple commits to make it easier to review:
* 08bd020: Replace all references to `React.PropTypes` with `prop-types` to avoid triggering our own warning message.
* ef5b5c6: Removed several references to `React.createClass` that appeared after rebasing this branch. (reviewed by @flarnie)
* 524ce20: Added getters for `createClass` and `PropTypes` to the main `React` isomorphic object, behind one-time warning messages. (reviewed by @spicyj)
* db48f54: Fixed Rollup bundles to inline 'prop-types' and 'create-react-class' for UMD builds only. (reviewed by @spicyj, @trueadm )
* cf49cfd: Updated `tests-passing.txt` to remove tests that were deleted in this branch.
* d34109a: Responses to PR feedback from @spicyj. (Added `package.json` dependencies to packages/react and packages/react-dom. Renamed a var. Expanded on an inline comment.)
* 488c8d2: Added warning for moved package to `React.checkPropTypes` accessor too and updated build script.
* 83bcb29: Wordsmithing for deprecation notices (added fb.me links).
* afdc9d2: Tweaked legacy module inlining to remove order-of-deps constraint
* d1348b9: Removed `$FlowFixMe`.
* 7dbc3e7: More wordsmithing of deprecation notices based on Dan's feedback.

### CODE REVIEW

**Comment by bvaughn** [comment]
Should be ready for review, @acdlite

**Comment by flarnie** [comment]
"Fixed several createClass references in fixtures" lgtm - reviewing the other change now.

File: package.json [hunk 1]
```diff
@@ -95,6 +93,10 @@
     "node": "4.x || 5.x || 6.x || 7.x",
     "npm": "2.x || 3.x || 4.x"
   },
+  "dependencies": {
```
Comment (spicyj) [inline package.json]: Why did you move these? I think dev should be fine. This package.json is only used for development; we have others for the npm packages that get released.

File: scripts/rollup/bundles.js [hunk 2]
```diff
@@ -45,7 +45,12 @@ const bundles = [
       sourceMap: false,
     },
     entry: 'src/isomorphic/React.js',
-    externals: [],
+    externals: [
+      'create-react-class/factory',
+      'prop-types',
+      'prop-types/checkPropTypes',
```
Comment (spicyj) [inline scripts/rollup/bundles.js]: Does only isomorphic use checkPropTypes like this? I guess I would have thought that context type checking does too.

File: src/isomorphic/React.js [hunk 3]
```diff
@@ -106,7 +110,19 @@ if (__DEV__) {
...
         warnedForCreateClass = true;
-        return undefined;
+        return createReactClass;
+      },
+    });
+
+    Object.defineProperty(React, 'PropTypes', {
+      get() {
+        warning(
+          didWarnPropTypesDeprecated,
```
Comment (spicyj) [inline src/isomorphic/React.js]: nit: inconsistent naming with this and warnedForCreateClass

File: scripts/rollup/modules.js [hunk 4]
```diff
@@ -52,6 +52,17 @@ const devOnlyFilesToStubOut = [
   "'ReactTestUtils'",
 ];
 
+// Ordering of these imports is important;
+// The default import must follow deep imports or Rollup breaks.
```
Comment (spicyj) [inline scripts/rollup/modules.js]: Why does Rollup break? (This seems brittle.)

File: package.json (hunk 1 above)
Comment (bvaughn) [inline package.json]: D'oh! Habit. I'll undo _this_ change and update the `packages/*/package.json` files instead 😄

File: scripts/rollup/bundles.js (hunk 2 above)
Comment (bvaughn) [inline scripts/rollup/bundles.js]: DOM does check too but it uses `React.checkPropTypes`. Rollup also only complained about the externals reference for isomorphic.

File: src/isomorphic/React.js (hunk 3 above)
Comment (bvaughn) [inline src/isomorphic/React.js]: 👍

File: scripts/rollup/modules.js (hunk 4 above)
Comment (bvaughn) [inline scripts/rollup/modules.js]: I gave an example below (in the comments) of what it maps the requires to but I'll update it to be a bit more explicit. Basically if it partial-matches one of the keys (eg `prop-types/checkPropTypes` matches `prop-types`) then it stops searching and uses a path like `prop-types/index.js/checkPropTypes.js`

File: scripts/rollup/modules.js [hunk 5]
```diff
@@ -230,6 +241,27 @@ function replaceDevOnlyStubbedModules(bundleType) {
   }
 }
 
+function replaceLegacyModuleAliases(bundleType) {
+  switch (bundleType) {
+    case UMD_DEV:
+    case UMD_PROD:
+      const modulesAlias = {};
+      for (var legacyModule in legacyModules) {
```
Comment (trueadm) [inline scripts/rollup/modules.js]: nit: can we use `const` or `let` here?

Comment (bvaughn) [inline scripts/rollup/modules.js]: Yup!

File: src/isomorphic/React.js (hunk 3 above)
Comment (acdlite) [inline src/isomorphic/React.js]: Let's add something like "it will be removed completely in React 16"

Also, maybe we could make this warning and the one for `React.createClass` more consistent.

File: src/isomorphic/React.js [hunk 6]
```diff
@@ -106,7 +110,19 @@ if (__DEV__) {
...
+        return createReactClass;
+      },
+    });
+
+    Object.defineProperty(React, 'PropTypes', {
+      get() {
+        warning(
+          warnedForPropTypes,
+          'PropTypes have moved out of the react package. ' +
+            'Use the prop-types package from npm instead.',
```
Comment (acdlite) [inline src/isomorphic/React.js]: See https://github.com/facebook/react/pull/9399#discussion_r110997070

Comment (acdlite) [inline src/isomorphic/React.js]: Suggestion:

> PropTypes have been moved to a separate package. Accessing React.PropTypes is no longer supported, and will be removed completely in React 16. Use the prop-types package on npm instead.

Comment (acdlite) [inline src/isomorphic/React.js]: And then with a fb.me link

File: scripts/rollup/modules.js (hunk 4 above)
Comment (gaearon) [inline scripts/rollup/modules.js]: Doesn't this mean that the replacement logic misses `'` or something that delimits end of the string, and we should add that?

File: src/isomorphic/React.js (hunk 6 above)
Comment (bvaughn) [inline src/isomorphic/React.js]: Sweet. Will do.

File: scripts/rollup/modules.js (hunk 4 above)
Comment (gaearon) [inline scripts/rollup/modules.js]: cc @trueadm

**Comment by bvaughn** [comment]
Added wordsmithing suggestions in 83bcb29 @acdlite

File: scripts/rollup/modules.js [hunk 7]
```diff
@@ -230,6 +241,27 @@ function replaceDevOnlyStubbedModules(bundleType) {
...
+        modulesAlias[legacyModule] = resolve(
```
Comment (trueadm) [inline scripts/rollup/modules.js]: As per @gaearon's point above, if we put quotes around this, so it's `moduleAlias[\`'${legacyModule}'\`]` then the legacyModule order shouldn't matter as the replace module literally "just" does string replacement :)

File: scripts/rollup/modules.js (hunk 4 above)
Comment (trueadm) [inline scripts/rollup/modules.js]: @gaearon I added a comment below, we don't need this logic here :) this should address what @spicyj said as well

File: scripts/rollup/modules.js (hunk 7 above)
Comment (bvaughn) [inline scripts/rollup/modules.js]: Nice! I'll try that. 😁

Comment (bvaughn) [inline scripts/rollup/modules.js]: Changed in afdc9d2.

**Comment by bvaughn** [comment]
Tagging @acdlite to review 488c8d2 and 83bcb29
Tagging @trueadm to review afdc9d2

File: src/isomorphic/React.js [hunk 8]
```diff
@@ -106,8 +106,11 @@ if (__DEV__) {
       get() {
         warning(
           warnedForCheckPropTypes,
-          'checkPropTypes has moved out of the react package. ' +
-            'Use the prop-types package from npm instead.',
+          'checkPropTypes have been moved to a separate package. ' +
```
Comment (gaearon) [inline src/isomorphic/React.js]: Not "has been"?

File: src/isomorphic/__tests__/React-test.js [hunk 9]
```diff
@@ -35,10 +35,11 @@ describe('React', () => {
     expect(createClass).not.toBe(undefined);
     expectDev(console.error.calls.count()).toBe(1);
     expectDev(console.error.calls.argsFor(0)[0]).toContain(
-      'React.createClass is no longer supported. Use a plain ' +
-        "JavaScript class instead. If you're not yet ready to migrate, " +
+      'React.createClass is no longer supported. Use a plain JavaScript ' +
+        "class instead. If you're not yet ready to migrate, " +
         'create-react-class is available on npm as a temporary, ' +
-        'drop-in replacement.',
+        'drop-in replacement. ' +
```
Comment (gaearon) [inline src/isomorphic/__tests__/React-test.js]: Should we make it clear it will keep on working indefinitely? "temporary" doesn't sound very reassuring.

Comment (bvaughn) [inline src/isomorphic/__tests__/React-test.js]: Yeah totally. I'll remove the word "temporary"

File: src/isomorphic/React.js (hunk 8 above)
Comment (bvaughn) [inline src/isomorphic/React.js]: Words are hard 😅
